main.cpp

```diff
@@ -851,15 +870,8 @@ struct router_thread
 		ios.stop();
 	}
 
-	void start(int threadid)
+	void start()
 	{
-		if (socks.empty())
-		{
-			fprintf(stderr, "no interfaces to receive on\n");
-			return;
-		}
-
-		this->threadid = threadid;
```

Comment:
I moved the thread id around a bit here, to have it be available in the constructor, in order to create unique names for the mapped files